Legobot/Connectors/Slack.py

```diff
@@ -170,8 +170,10 @@ def _parse_metadata(self, message):
         elif 'bot_id' in message:
             metadata['source_user'] = message['bot_id']
 
-        metadata['user_id'] = metadata['source_user']
-        metadata['display_name'] = self.get_username(metadata['source_user'])
+        if 'source_user' in metadata:
```

Comment:
What happens if a lego depends on `source_user`?

Reply:
it's out of luck?
to be clear we weren't accommodating for this use case previously, simply ignoring it. With the addition of trying to convert source_user to these other metadata pieces, the missing source_user caused the bot to crash. This is an attempt to return to status quo (ignoring) not to accommodate the odd messages. If you want to kick it back I'll look for a full remedy.

Reply:
Base fields should be a `None` when they can't be filled, so we should add this at the parent Lego level. The idea is that this is part of the standard set of metadata and we don't want legos to throw a key error when looking for it.